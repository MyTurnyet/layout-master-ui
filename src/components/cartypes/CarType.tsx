import React, {FunctionComponent} from "react";
import AarDesignation from "./AarDesignation";
import CarriedGoodsList from "./CarriedGoodsList";

export type CarTypeProps = {
    aarDesignation: string;
    carriedGoodsList: string[];
    id: string;
}

const CarType: FunctionComponent<CarTypeProps> = (props: CarTypeProps & { children?: React.ReactNode })=>{
    return (<div className={`CarType`} id={props.id}>
        <AarDesignation initials={props.aarDesignation}/>
        <CarriedGoodsList goods={props.carriedGoodsList}/>
    </div>);
}
export default CarType;
