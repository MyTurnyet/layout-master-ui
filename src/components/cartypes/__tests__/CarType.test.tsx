import React from 'react';
import {render, screen} from '@testing-library/react';
import CarType, {CarTypeProps} from "../CarType";

describe(' CarType ', () => {
    it('should render with AarType ', () => {
        const carType: CarTypeProps = {
            aarDesignation: "GS",
            carriedGoodsList: ["Logs", "Parts"],
            id: "12345"
        }
        render(<CarType {...carType} />);
        const aarElement = screen.getByText("GS");
        expect(aarElement).toBeInTheDocument()
        expect(aarElement).toHaveClass("AarDesignation")
        expect(aarElement.parentElement).toHaveClass("CarType")

        const goodsElement = screen.getByText("Logs, Parts");
        expect(goodsElement).toBeInTheDocument()
         expect(goodsElement).toHaveClass("CarriedGoodsList")
         expect(goodsElement.parentElement).toHaveClass("CarType")
    })
});
