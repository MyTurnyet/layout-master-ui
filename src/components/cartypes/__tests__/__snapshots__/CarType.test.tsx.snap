// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` CarType  should render with AarType  1`] = `
<div
  class="sc-dlfnbm iWniRb"
  id="12345"
>
  <div
    class="sc-bdfBwQ bBNLzT"
  >
    GS
  </div>
  <div
    class="sc-gsTCUz wJjXs"
  >
    Logs, Parts
  </div>
</div>
`;
